modules/core/shared/src/main/scala-3/io/circe/derivation/ConfiguredDecoder.scala

```diff
@@ -53,7 +47,14 @@ trait ConfiguredDecoder[A](using conf: Configuration) extends Decoder[A]:
                 cursor.history
               )
             )
-          case Right(Some(sumTypeName)) => fromName(sumTypeName, c)
+          case Right(Some(sumTypeName)) =>
+            decodersDict
+              .get(sumTypeName)
+              .fold(
+                nameNotFound(sumTypeName, c)
+              ) { decoder =>
+                decode(decoder.asInstanceOf[Decoder[A]])(c)
+              }
```

Comment:
FYI: this is basically the implementation of `def fromName` from previous line 35 ff.